src/main/java/adris/altoclef/util/WorldUtil.java

```diff
@@ -87,57 +107,61 @@ static boolean isChestBig(AltoClef mod, BlockPos posWithChest) {
         for (int y = 255; y >= 0; --y) {
             BlockPos check = new BlockPos(x, y, z);
-            if (possibleBlocks.contains(mod.getWorld().getBlockState(check).getBlock())) return y;
+            if (possibleBlocks.contains(mod.getWorld().getBlockState(check).getBlock()))
+                return y;
 
         }
         return -1;
     }
 
-    static boolean canBreak(AltoClef mod, BlockPos pos) {
-        return mod.getWorld().getBlockState(pos).getHardness(mod.getWorld(), pos) >= 0 && !mod.getExtraBaritoneSettings().shouldAvoidBreaking(pos);
+    public static boolean canBreak(AltoClef mod, BlockPos pos) {
+        return mod.getWorld().getBlockState(pos).getHardness(mod.getWorld(), pos) >= 0 &&
+               !mod.getExtraBaritoneSettings().shouldAvoidBreaking(pos);
     }
-    static boolean canPlace(AltoClef mod, BlockPos pos) {
+
+    public static boolean canPlace(AltoClef mod, BlockPos pos) {
         return !mod.getExtraBaritoneSettings().shouldAvoidPlacingAt(pos);
     }
 
-    static boolean isAir(AltoClef mod, BlockPos pos) {
+    public static boolean isAir(AltoClef mod, BlockPos pos) {
         return mod.getBlockTracker().blockIsValid(pos, Blocks.AIR, Blocks.CAVE_AIR, Blocks.VOID_AIR);
         //return state.isAir() || isAir(state.getBlock());
     }
-    static boolean isAir(Block block) {
+
+    public static boolean isAir(Block block) {
         return block == Blocks.AIR || block == Blocks.CAVE_AIR || block == Blocks.VOID_AIR;
     }
 
-    static boolean isContainerBlock(AltoClef mod, BlockPos pos) {
+    public static boolean isContainerBlock(AltoClef mod, BlockPos pos) {
         Block block = mod.getWorld().getBlockState(pos).getBlock();
-        return (block instanceof ChestBlock
-                || block instanceof EnderChestBlock
-                || block instanceof CraftingTableBlock
-                || block instanceof AbstractFurnaceBlock
-                || block instanceof LoomBlock
-                || block instanceof CartographyTableBlock
-                || block instanceof EnchantingTableBlock
-        );
+        return (block instanceof ChestBlock || block instanceof EnderChestBlock || block instanceof CraftingTableBlock ||
+                block instanceof AbstractFurnaceBlock || block instanceof LoomBlock || block instanceof CartographyTableBlock ||
+                block instanceof EnchantingTableBlock);
     }
 
-    static boolean isInsidePlayer(AltoClef mod, BlockPos pos) {
+    public static boolean isInsidePlayer(AltoClef mod, BlockPos pos) {
         return pos.isWithinDistance(mod.getPlayer().getPos(), 2);
     }
 
-    static Iterable<BlockPos> scanRegion(AltoClef mod, BlockPos start, BlockPos end) {
+    public static Iterable<BlockPos> scanRegion(AltoClef mod, BlockPos start, BlockPos end) {
         return () -> new Iterator<BlockPos>() {
-            int x = start.getX(), y = start.getY(), z = start.getZ();
+            int x = start.getX(); // TODO: 2021-05-22 what the shit?
```

Comment:
oof, I wanted to make an iterator that iterates through a 3d box but didn't know the standard way to do this in java.

python and C# lets you yield like a coroutine, java has iterables.

So I wanted the java equivalent of

```java
IterableEnumerableWhatever scanArea(BlockPos start, BlockPos end) {
    for (x = start.x; x <= end.x; ++x) {
        for (y = start.y; y <= end.y; ++y) {
            for (z = start.z; z <= end.z; ++z) {
                yield return new BlockPos(x,y,z);
            }
        }
    }
}
```

Reply:
What exactly is it that you're trying to do? Because afaik, that could be done with [normal for loops.

Reply:
Ok so, there were multiple cases where I wanted to iterate through an X by Y by Z area.

I could do this every time;
```java
for (x = start.x; x <= end.x; ++x) {
        for (y = start.y; y <= end.y; ++y) {
            for (z = start.z; z <= end.z; ++z) {
                BlockPos pos = new BlockPos(x, y, z);
                // Do something with pos.
            }
        }
    }
```
But that is kinda annoying. I'd much rather the code look like this:
```java
for (BlockPos pos : blockRange(start, end)) {
    // Do something with pos.
}
```